### PR TITLE
WIP fix: prevent error in util:get-document-by-absolute-id

### PR DESCRIPTION
**Pull request by line-o**
### Description:

`util:get-document-by-absolute-id` returns an empty sequence if document was not found instead of throwing an error.

### Reference:

none

### Type of tests:

TBD

### CODE REVIEW

File: src/org/exist/storage/NativeBroker.java [hunk 1]
```diff
@@ -2021,6 +2021,9 @@ public DocumentImpl getResourceById(final int collectionId, final byte resourceT
             //get the resource uri
             final Value key = new CollectionStore.DocumentKey(collectionId, resourceType, documentId);
             final VariableByteInput vbi = collectionsDb.getAsStream(key);
+            if (vbi == null) {
```
Comment (adamretter) [inline src/org/exist/storage/NativeBroker.java]: I am not sure that this change is meant to be in this PR? It doesn't seem to be related to the topic of the PR. Can you explain please?

File: src/org/exist/xquery/functions/util/DocumentNameOrId.java [hunk 2]
```diff
@@ -163,6 +163,9 @@ public Sequence eval(Sequence[] args, Sequence contextSequence) throws XPathExce
                     final int collectionId = absoluteId.and(BigInteger.valueOf(0xFFFFFFFF)).intValue();
                     
                     doc = context.getBroker().getResourceById(collectionId, resourceType, documentId);
+                    if (doc == null) {
```
Comment (adamretter) [inline src/org/exist/xquery/functions/util/DocumentNameOrId.java]: Looks good. I think it would also make sense to add Javadoc on `DBBroker#getResourceById(int, byte, int)` so that it documents that the return type might be null. Would you be able to do that too please?

File: src/org/exist/storage/NativeBroker.java [hunk 3]
```diff
@@ -2021,6 +2021,9 @@ public DocumentImpl getResourceById(final int collectionId, final byte resourceT
...
+                return null;
+            }
             vbi.readInt(); //skip doc id
```
Comment (line-o) [inline src/org/exist/storage/NativeBroker.java]: This is the actual code that throws. So we must return null beforehand.

File: src/org/exist/storage/NativeBroker.java (hunk 1 above)
Comment (line-o) [reply src/org/exist/storage/NativeBroker.java]: It is necessary in order to avoid try catch around the call to 
```
 doc = context.getBroker().getResourceById(collectionId, resourceType, documentId);
```

File: src/org/exist/xquery/functions/util/DocumentNameOrId.java (hunk 2 above)
Comment (line-o) [reply src/org/exist/xquery/functions/util/DocumentNameOrId.java]: Will do that

**Comment by line-o** [comment]
As I am not allowed to push to this repo, I made the changes here
https://github.com/line-o/exist/tree/fix/get-document-by-absolute-id
Should I create a new PR?

File: src/org/exist/storage/NativeBroker.java (hunk 1 above)
Comment (adamretter) [reply src/org/exist/storage/NativeBroker.java]: Hmm... doesn't a `null` here imply that the `collections.dbx` has become corrrupted?

File: src/org/exist/storage/NativeBroker.java (hunk 3 above)
Comment (adamretter) [reply src/org/exist/storage/NativeBroker.java]: Hmm... doesn't a null here imply that the collections.dbx has become corrrupted?

**Comment by adamretter** [comment]
I understand that @wolfgangmm is addressing with @line-o

**Comment by adamretter** [comment]
@line-o @wolfgangmm I am going to close this as no progress is being made. Please feel free to re-open if you have new stuff
